src/logic/map.cc

```diff
@@ -2092,8 +2092,9 @@ std::unique_ptr<MapLoader> Map::get_correct_loader(const std::string& filename)
 		if (boost::algorithm::ends_with(lower_filename, kWidelandsMapExtension)) {
 			result.reset(new WidelandsMapLoader(g_fs->make_sub_file_system(filename), this));
 		} else if (boost::algorithm::ends_with(lower_filename, kSavegameExtension)) {
-			result.reset(new WidelandsMapLoader(
-			   g_fs->make_sub_file_system(filename)->make_sub_file_system("map"), this));
+			FileSystem* sub_fs = g_fs->make_sub_file_system(filename);
```

Comment:
We're tying to avoid adding new raw deletes - please use a unique_ptr

Reply:
Ok, I'm fairly new to C++1x, so thanks for letting me know.

Reply:
You're welcome. I did not know any C++ when I first joined this project ;)